### DISCUSSION TITLE
Allow interleaving times across topics with websocket protocol

### DISCUSSION THREAD

**Discussion by jameskuszmaul-brt** [foxglove/community#310] [Q&A]
I would like to be able to have Studio subscribe to two websocket topics, e.g. `/a` and `/b`, and then have the following messages come in in this order, and without having the display cleared:

```
/a: timestamp 1.0 sec
/b: timestamp 0.5 sec
/a: timestamp 2.0 sec
```

And then if I have a 2-D plot or just am looking at raw messages of `/a` and `/b` I would like to have everything still displayed.

Essentially, I am happy to have the server guarantee that an individual topic will send all of its messages across the websocket in order (and I think that's consistent with most pub-sub frameworks out there), but it is much more obnoxious to guarantee that messages across *all* topics are sorted before sending them across.

**Comment by pickledgator** [comment]
+1!
